src/BLEPeripheral.cpp

```diff
@@ -69,39 +69,54 @@ BLEPeripheral::~BLEPeripheral() {
 }
 
 void BLEPeripheral::begin() {
-  unsigned char advertisementDataType = 0;
   unsigned char scanDataType = 0;
 
-  unsigned char advertisementDataLength = 0;
+  unsigned char advertisementDataSize = 0;
   unsigned char scanDataLength = 0;
 
-  unsigned char advertisementData[BLE_ADVERTISEMENT_DATA_MAX_VALUE_LENGTH];
+  BLEAdvertisementData advertisementData[3];
```

Comment:
I can only think of having the size of this array in DeviceLimits so it could be different for different boards, setting it to 1 on nRF8001, but I wouldn't know what it should be on nRF51822.